x-pack/plugins/security_solution/cypress/e2e/detection_rules/persistent_rules_table_state.cy.ts

```diff
@@ -0,0 +1,356 @@
+        changeRulesTableState();
+        goToTablePage(2);
+
+        cy.reload();
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(2);
+      });
+
+      it('persists after navigating back from a rule details page', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        goToRuleDetails();
+        cy.go('back');
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(2);
+      });
+
+      it('persists after navigating to another page inside Security Solution', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        visit(DASHBOARDS_URL);
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(1);
+      });
+
+      it('persists after navigating to another page outside Security Solution', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        visit(KIBANA_HOME);
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(1);
+      });
+    });
+
+    describe('and on the rules monitoring tab', () => {
+      beforeEach(() => {
+        visit(SECURITY_DETECTIONS_RULES_MONITORING_URL);
+
+        changeRulesTableState();
+        goToTablePage(2);
+      });
+
+      it('persists the selected tab', () => {
+        cy.reload();
+
+        expectRulesMonitoringTab();
+      });
+    });
+  });
+
+  describe('upon state format upgrade', async () => {
+    describe('and having state in the url', () => {
+      it('ignores unsupported state key', () => {
+        visitRulesTableWithState({
+          someKey: 10,
+          searchTerm: 'rule',
+          tags: ['tag-b'],
+          source: 'custom',
+          enabled: false,
+          field: 'name',
+          order: 'asc',
+          perPage: 5,
+          page: 2,
+        });
+
+        expectRulesTableState();
+        expectTablePage(2);
+        expectManagementTableRules(['rule 6']);
+      });
+    });
+
+    describe('and having state in the session storage', () => {
+      it('ignores unsupported state key', () => {
+        setStorageState({
+          someKey: 10,
+          searchTerm: 'rule',
+          tags: ['tag-b'],
+          source: 'custom',
+          enabled: false,
+          field: 'name',
+          order: 'asc',
+          perPage: 5,
+        });
+
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        expectRulesTableState();
+        expectTablePage(1);
+      });
+    });
+  });
+
+  describe('when persisted state is partially unavailable', () => {
+    describe('and on the rules management tab', () => {
+      beforeEach(() => {
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        changeRulesTableState();
+        goToTablePage(2);
```

Comment:
Same as https://github.com/elastic/kibana/pull/149638/files#r1106910760, let's move these steps to test bodies.